etcdserver/api/v3rpc/rpctypes/error.go

```diff
@@ -20,28 +20,28 @@ import (
+	ErrFutureRev    = rpcError{s: grpc.Errorf(codes.OutOfRange, "etcdserver: mvcc: required revision is a future revision").Error()}
+	ErrNoSpace      = rpcError{s: grpc.Errorf(codes.ResourceExhausted, "etcdserver: mvcc: database space exceeded").Error()}
 
-	ErrLeaseNotFound = grpc.Errorf(codes.NotFound, "etcdserver: requested lease not found")
-	ErrLeaseExist    = grpc.Errorf(codes.FailedPrecondition, "etcdserver: lease already exists")
+	ErrLeaseNotFound = rpcError{s: grpc.Errorf(codes.NotFound, "etcdserver: requested lease not found").Error()}
+	ErrLeaseExist    = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: lease already exists").Error()}
 
-	ErrMemberExist    = grpc.Errorf(codes.FailedPrecondition, "etcdserver: member ID already exist")
-	ErrPeerURLExist   = grpc.Errorf(codes.FailedPrecondition, "etcdserver: Peer URLs already exists")
-	ErrMemberBadURLs  = grpc.Errorf(codes.InvalidArgument, "etcdserver: given member URLs are invalid")
-	ErrMemberNotFound = grpc.Errorf(codes.NotFound, "etcdserver: member not found")
+	ErrMemberExist    = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: member ID already exist").Error()}
+	ErrPeerURLExist   = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: Peer URLs already exists").Error()}
+	ErrMemberBadURLs  = rpcError{s: grpc.Errorf(codes.InvalidArgument, "etcdserver: given member URLs are invalid").Error()}
+	ErrMemberNotFound = rpcError{s: grpc.Errorf(codes.NotFound, "etcdserver: member not found").Error()}
 
-	ErrRequestTooLarge = grpc.Errorf(codes.InvalidArgument, "etcdserver: request is too large")
+	ErrRequestTooLarge = rpcError{s: grpc.Errorf(codes.InvalidArgument, "etcdserver: request is too large").Error()}
 
-	ErrUserAlreadyExist = grpc.Errorf(codes.FailedPrecondition, "etcdserver: user name already exists")
-	ErrUserNotFound     = grpc.Errorf(codes.FailedPrecondition, "etcdserver: user name not found")
-	ErrRoleAlreadyExist = grpc.Errorf(codes.FailedPrecondition, "etcdserver: role name already exists")
-	ErrRoleNotFound     = grpc.Errorf(codes.FailedPrecondition, "etcdserver: role name not found")
-	ErrAuthFailed       = grpc.Errorf(codes.InvalidArgument, "etcdserver: authentication failed, invalid user ID or password")
+	ErrUserAlreadyExist = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: user name already exists").Error()}
+	ErrUserNotFound     = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: user name not found").Error()}
+	ErrRoleAlreadyExist = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: role name already exists").Error()}
+	ErrRoleNotFound     = rpcError{s: grpc.Errorf(codes.FailedPrecondition, "etcdserver: role name not found").Error()}
+	ErrAuthFailed       = rpcError{s: grpc.Errorf(codes.InvalidArgument, "etcdserver: authentication failed, invalid user ID or password").Error()}
```

Comment:
Ok, how about just leave it as grpcError type and use the conversion function ONLY IN CLIENT side?